### DISCUSSION TITLE
Which lib should I include to be able to use SimpleHexAtoi?

### DISCUSSION THREAD

**Discussion by sborisov-int** [abseil/abseil-cpp#1714] [General]
Hi,

Thanks for a great library!

I'm trying to use SimpleHexAtoi() for int128, but get the following error message:
`Severity	Code	Description	Project	File	Line	Suppression State	Details
Error	LNK2019	unresolved external symbol "bool __cdecl absl::numbers_internal::safe_strto128_base(class std::basic_string_view<char,struct std::char_traits<char> >,class absl::int128 *,int)" (?safe_strto128_base@numbers_internal@absl@@YA_NV?$basic_string_view@DU?$char_traits@D@std@@@std@@PEAVint128@2@H@Z) referenced in function "bool __cdecl absl::SimpleHexAtoi(class std::basic_string_view<char,struct std::char_traits<char> >,class absl::int128 *)" (?SimpleHexAtoi@absl@@YA_NV?$basic_string_view@DU?$char_traits@D@std@@@std@@PEAVint128@1@@Z)	floating_point	`

Currently I included:
- absl_int128.lib
- absl_throw_delegate.lib
- absl_raw_logging_internal.lib
- absl_str_format_internal.lib
- absl_strings.lib
- absl_string_view.lib
- absl_flags_marshalling.lib

I'm using VS2022. Please help!

Thanks,
Slava

**Comment by derekmauro** [comment]
You need to link `absl_strings.lib` and all of its dependencies. Since you are already linking it, from the error message, I'm guessing the problem is that you have a mixed-mode build. Please see https://github.com/abseil/abseil-cpp/blob/master/FAQ.md#how-to-i-set-the-c-dialect-used-to-build-abseil.

**Comment by sborisov-int** [comment]
I tried to build this library as a dll and saw that SimpleHexAtoi does not exist in the .def file, so it is not exported. Moreover, I would move its implementation from .h to .cc and remove "inline". What do you think? Is there any unit test for calling SimpleHexAtoi from a lib or dll?

**Reply by derekmauro** [reply]
If you want further help, you need to post complete reproduction instructions.
